src/components/Input/label/index.js

```diff
@@ -29,6 +29,7 @@ Label.propTypes = {
     required: PropTypes.bool,
     inputId: PropTypes.string,
     readOnly: PropTypes.bool,
+    disabled: PropTypes.bool,
```

Comment:
To change the API of a component a previous discussion is necessary, this small change has a big impact, you had to change 5 component which in turn many more components depends on them for this reason it is not feasible. I think the best solution would be to add to the RenderIf where the Label is painted the conditions of disabled and readOnly